GTViewController/Classes/Custom Classes/GTFollowupThankYouView.m

```diff
@@ -0,0 +1,96 @@
+//
+//  GTFollowupThankYouView.m
+//  GTViewController
+//
+//  Created by Ryan Carlson on 4/12/16.
+//  Copyright © 2016 Michael Harrison. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+#import "GTFollowupThankYouView.h"
+#import "GTLabel.h"
+
+#import "GTPageInterpreter.h"
+#import "UISnuffleButton.h"
+
+@implementation GTFollowupThankYouView
+
+- (instancetype) initFromElement:(TBXMLElement *) thankYouComponentElement withFrame:(CGRect)frame withPageStyle:(GTPageStyle *)style {
+    
+    self = [super initWithFrame:frame];
+    [self setBackgroundColor:style.backgroundColor];
+    
+    TBXMLElement *thankYouElement = thankYouComponentElement->firstChild;
+    
+    int numLabels = 0;
+    int numTextInputs = 0;
+    int numButtonPairs = 0;
+    
+    CGFloat topLeadingSpace = 0;
+    CGFloat betweenElementsSpace = 0;
+    
+    CGFloat totalVerticalSpace = frame.size.height;
+    CGFloat totalUsedSpace = 0;
+
+    // first pass - count objects
+    while (thankYouElement) {
+        NSString *thankYouElementName = [TBXML elementName:thankYouElement];
+        
+        if ([thankYouElementName isEqual:kName_Label]) {
+            numLabels++;
+            totalUsedSpace += DEFAULT_HEIGHT_LABEL;
+        } else if ([thankYouElementName isEqual:kName_Button]) {
+            numLabels++;
+            totalUsedSpace += DEFAULT_HEIGHT_LABEL;
+        }
+        
+        thankYouElement = thankYouElement->nextSibling;
+    }
+    
+    // there are too many elements, not sure what to do
```

Comment:
I would just always lay it out the same. If there are too many it will overlap and look weird, which then puts it on the content creator to reduce the number of elements to make it look normal again. Usually by making an extra page